monai/data/dataset.py

```diff
@@ -337,27 +344,17 @@ def _cachecheck(self, item_transformed):
         _item_transformed = self._pre_transform(deepcopy(item_transformed))  # keep the original hashed
         if hashfile is None:
             return _item_transformed
-        try:
```

Comment:
Not sure if we should unify the saving function here, saving the model state and saving the cache would be very different

Reply:
Sure, restored the `PersistentDataset` in the latest commit.

Thanks.